assets/src/amp-story-editor-blocks.js

```diff
@@ -33,7 +55,23 @@ domReady( () => {
 } );
 
 subscribe( () => {
-	setDefaultBlockName( getSelectedBlockClientId() ? 'amp/amp-story-text' : 'amp/amp-story-page' );
+	const defaultBlockName = getDefaultBlockName();
+	const selectedBlockClientId = getSelectedBlockClientId();
+
+	// Ensure that the default block is page when no block is selected.
+	if ( selectedBlockClientId && 'amp/amp-story-text' !== defaultBlockName ) {
+		setDefaultBlockName( 'amp/amp-story-text' );
```

Comment:
Perhaps related but just checked that at this moment it's not possible to add a Page -- the `+` icon (inserter) does not show if a Page is selected, nor does the inserter below the Page show up if a Page is selected, at the same time it seems to be possible to add a Page inside a Page.
![screen shot 2019-03-01 at 4 07 20 pm](https://user-images.githubusercontent.com/3294597/53646777-54356300-3c3c-11e9-95fb-8b8d34c65310.png)

Reply:
> If the currently select block is an `amp/amp-story-page` then shouldn't the default block also be `amp/amp-story-page` instead of a text block?

Indeed! I might have forgotten about this scenario when optimizing the subscriber.

Reply:
> At the same time it seems to be possible to add a Page inside a Page.

Looks like `ALLOWED_BLOCKS` is used as the whitelist of inner blocks, which of contains pages as well.

Reply:
Let's further investigate this in #1917